monitor/consul/watcher.go

```diff
@@ -39,24 +38,20 @@ type Watcher struct {
 }
 
 // New creates a new watcher.
-func New(addr, dc, token string, timeout time.Duration, ii ...Item) (*Watcher, error) {
+func New(addr, dc, token string, ii ...Item) (*Watcher, error) {
 	if addr == "" {
 		return nil, errors.New("address is empty")
 	}
 	if len(ii) == 0 {
 		return nil, errors.New("items are empty")
 	}
-	if timeout == 0 {
-		timeout = 60 * time.Second
-	}
 	cfg := api.DefaultConfig()
 	cfg.Address = addr
 	var err error
 	cfg.HttpClient, err = api.NewHttpClient(cfg.Transport, cfg.TLSConfig)
 	if err != nil {
 		return nil, err
 	}
-	cfg.HttpClient.Timeout = timeout
```

Comment:
We can exchange this line with:
`cfg.WaitTime = timeout`
Which controls long pollint timeout.
Since we do not neet to set a timeout anymore, we can remove lines 54-58 also  and use the internally created http client.